### PR TITLE
added a user model and logic for registering a new user 

### PR DESCRIPTION
**Pull request by nshtykh**

### CODE REVIEW

File: src/main/java/com/example/onlinebookstore/dto/user/UserRegistrationRequestDto.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import org.hibernate.validator.constraints.Length;
+
+@Data
+@FieldMatch
+public class UserRegistrationRequestDto {
+    @NotNull
+    @Email
+    private String email;
+    @NotNull
+    @Length(min = 6, max = 20)
```
Comment (vovasalyha) [inline src/main/java/com/example/onlinebookstore/dto/user/UserRegistrationRequestDto.java]: Better use `Size` annotation, to be consistent. `Length` has the same meaning, but it's hibernate specific annotation.

File: src/main/java/com/example/onlinebookstore/dto/user/UserRegistrationRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import jakarta.validation.constraints.NotNull;
+import lombok.Data;
+import org.hibernate.validator.constraints.Length;
+
+@Data
+@FieldMatch
+public class UserRegistrationRequestDto {
+    @NotNull
+    @Email
+    private String email;
```
Comment (vovasalyha) [inline src/main/java/com/example/onlinebookstore/dto/user/UserRegistrationRequestDto.java:15]: Please validate text fields length, to not exceed DB column restrictions.

File: src/main/java/com/example/onlinebookstore/dto/user/UserRegistrationRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    @NotNull
+    @Length(min = 6, max = 20)
+    private String password;
+    @NotNull
+    @Length(min = 6, max = 20)
+    private String repeatPassword;
+    @NotNull
+    private String firstName;
```
Comment (vovasalyha) [inline src/main/java/com/example/onlinebookstore/dto/user/UserRegistrationRequestDto.java:24]: What if I pass an empty string here? Or if I pass a string which is longer that maximum column size in DB?

File: src/main/java/com/example/onlinebookstore/dto/user/UserRegistrationRequestDto.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    @NotNull
+    private String lastName;
```
Comment (vovasalyha) [inline src/main/java/com/example/onlinebookstore/dto/user/UserRegistrationRequestDto.java:27]: What if I pass an empty string here? Or if I pass a string which is longer that maximum column size in DB?

File: src/main/java/com/example/onlinebookstore/exception/CustomGlobalExceptionHandler.java [hunk 5]
```diff
@@ -44,4 +45,22 @@ private String getErrorMessage(ObjectError e) {
...
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", HttpStatus.BAD_REQUEST);
+        body.put("errors", exception.getMessage());
+        return new ResponseEntity<>(body, HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler(EntityNotFoundException.class)
+    protected ResponseEntity<Object> handleEntityNotFoundException(Exception exception) {
+        Map<String, Object> body = new LinkedHashMap<>();
```
Comment (vovasalyha) [inline src/main/java/com/example/onlinebookstore/exception/CustomGlobalExceptionHandler.java:62]: A lot of code duplication. Create a separate private method which builds the map.

File: src/main/java/com/example/onlinebookstore/service/user/UserServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto request) {
+        if (userRepository.findByEmail(request.getEmail()).isPresent()) {
+            throw new RegistrationException("Cant' register user!");
```
Comment (vovasalyha) [inline src/main/java/com/example/onlinebookstore/service/user/UserServiceImpl.java]:
```suggestion
            throw new RegistrationException("Can't register a user.");

```

**Review by vovasalyha — CHANGES_REQUESTED** [review]
Looks good, but see a couple of comments.

File: src/main/java/com/example/onlinebookstore/exception/CustomGlobalExceptionHandler.java (hunk 5 above)
Comment (dshulha) [reply src/main/java/com/example/onlinebookstore/exception/CustomGlobalExceptionHandler.java:62]: Now you send `BAD_REQUEST` status when `EntityNotFoundException` was thrown. But you should send `NOT_FOUND` status for this case. Also you can specify `@ExceptionHandler` by array of exceptions.

**Review by dshulha — CHANGES_REQUESTED** [review]
You should rewrite logic in your `CustomGlobalExceptionHandler.java` according to my comment

**Pull request merged**